### PR TITLE
feat : add to cart and wishlist

### PR DESCRIPTION
**Pull request by ajay-mandaviya**
add product into cart if user login
add to wishlist if user login
move to wishlist product into cart 
move to cart  product into wishlist

### CODE REVIEW

**Review by sharathnair9999 — APPROVED** [review]

File: src/components/NavBar/Nav.jsx [hunk 1]
```diff
@@ -43,6 +43,9 @@ const Nav = () => {
               Logout
             </button>
           ) : (
+            // <div className="badge-container">
+            //   <i class="fa-solid fa-user fa-2x"></i>
+            // </div>
```
Comment (prathmeshjagtap) [inline src/components/NavBar/Nav.jsx:48]: you can remove commented  code if you are not going  to use it in future

File: src/components/SingleProduct/SignleProduct.jsx [hunk 2]
```diff
@@ -18,37 +19,30 @@ const SignleProduct = ({ product }) => {
...
       navigate("/login");
     } else {
-      dispatch({
-        type: ADD_TO_WISHLIST,
-        payload: product,
-      });
+      addToWishList(dispatch, product, token);
     }
   };
+  // console.log("cart in ", cart);
```
Comment (prathmeshjagtap) [inline src/components/SingleProduct/SignleProduct.jsx:38]: remove console.log

File: src/components/SingleProduct/SignleProduct.jsx [hunk 3]
```diff
@@ -70,7 +64,7 @@ const SignleProduct = ({ product }) => {
             )}
           </div>
         </div>
-        {cart.some((cart) => cart.id === product.id) ? (
+        {cart.some((cart) => cart._id === product._id) ? (
           <button className="card-btn" onClick={() => {}}>
             Go to cart
```
Comment (prathmeshjagtap) [inline src/components/SingleProduct/SignleProduct.jsx:69]: you can navigate user to cart by using useNavigate from react-router

File: src/pages/Login/Login.jsx [hunk 4]
```diff
@@ -28,8 +28,7 @@ const Login = () => {
     console.log(userLogin);
     loginWithUser(userLogin);
```
Comment (prathmeshjagtap) [inline src/pages/Login/Login.jsx:29]: remove console.log

File: src/services/wishListApi.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        headers: {
+          authorization: token,
+        },
+      }
+    );
+    dispatch({
+      type: ADD_TO_WISHLIST,
+      payload: wishlist,
+    });
+    console.log("adding wishlist response ", wishlist);
```
Comment (prathmeshjagtap) [inline src/services/wishListApi.js:40]: you can remove this console.log

**Review by prathmeshjagtap — CHANGES_REQUESTED** [review]
code is Easily understandable,  but your preview is not working, I can't signup or Login from the preview link, I think you haven't  Set Environment Variables in Netlify. 

If you have not set up Environment Variables:-  Here is the path to do 
netlify -> your website -> site setting -> see on left side pannel (build and deploy ) -> same left side pannel -> environment -> add there and save

then you can clear and deploy again it will work.

**Pull request merged**